main.py

```diff
@@ -0,0 +1,78 @@
+import json
+from tkinter import *
+from tkinter import messagebox
+data_file = open("data.json")
+data = json.load(data_file)
+
+
+# save() gets input values from website, email, and password. Loads data file from json.
+#  appends new data to list, writes the updated list back to json file.
+def save():
+
+    website = website_entry.get()
+    email = email_entry.get()
+    password = password_entry.get()
+
+    if len(website) == data["MINIMUM_VALUE"] or len(email) == data["MINIMUM_VALUE"]\
+            or len(password) == data["MINIMUM_VALUE"]:
+        messagebox.showinfo(title=data["ERROR_TITLE"], message=data["ERROR_MESSAGE"])
+    else:
+        is_ok = messagebox.askokcancel(title=website, message=f"These are the details entered: \nEmail: {email} " 
+                                                      f"\nPassword: {password}")
+
+        # check if entered values are correct
+        if is_ok:
+            with open(data["SAVED_PASSWORDS_FILE"]) as fp:
+                stored_passwords = json.load(fp)
+
+            stored_passwords.append({
+                "Website": website,
+                "Email": email,
+                "Password": password
+            })
+
+            with open(data["SAVED_PASSWORDS_FILE"], 'w') as json_file:
+                json.dump(stored_passwords, json_file, indent=data["INDENT_VALUE"], separators=(',', ':'))
+                website_entry.delete(data["START_INDEX"], END)
+                email_entry.delete(data["START_INDEX"], END)
+                password_entry.delete(data["START_INDEX"], END)
+
+
+# window
+window = Tk()
+window.title(data["WINDOW_TITLE"])
+window.config(padx=data["WINDOW_PAD_X"], pady=data["WINDOW_PAD_Y"])
+
+# image
```

Comment:
A lot of this logic could be encapsulated within functions

Reply:
corrected. Due to the following code below calling the save function when clicking the add button. Without being able to pass parameters for the website_entry, email_entry, and password_entry variable, I had to create a Class to initialize these variables to allow them to be shared across functions

`add_button = Button(text=data["ADD_BUTTON_TEXT"], width=data["ADD_BUTTON_WIDTH"], command=### save)`